metagraph/tests/annotation/row_diff/test_row_diff.cpp

```diff
@@ -259,7 +264,8 @@ TEST(RowDiff, GetAnnotationBifurcationMasked) {
 
     annot::binmat::ColumnMajor mat(std::move(cols));
 
-    annot::binmat::RowDiff annot(&graph, std::move(mat), fterm_temp.name());
+    annot::binmat::RowDiff annot(&graph, std::move(mat));
+    annot.load_anchor(fterm_temp.name());
```

Comment:
Isn't it easier to pass anchor_filename to the constructor?

Reply:
Maybe add a parameter to the constructor, but make it false by default?
It's just a bit strange that you have to pass a filename but the file is not loaded. If there is no other parameter, everyone will assume that it loads the vector (the thought process is: I have to pass a filename, so that file must be loaded). If there is another parameter, then it will be clear that the thing is loaded only if the flag is set to true.

Reply:
Hmmm, not sure what you mean. The current constructor does not require passing in any filename. Or were you talking hypothetically, if I added a filename parameter?

Reply:
Ah, I had in my head the old version where you had to pass a filename.

So what is the scheme now then, row-diff columns don't know the location of their anchors?

Reply:
Right. `RowDiff<ColumnMajor>` is completely ignorant. Someone needs to tell it to load its anchors and where to load them from.
In contrast, as we discussed, `RowDiff<AnythingOther>` simply saves its anchors in the same file as the annotation.

Reply:
Ok, can we add this filename path to the constructor then?
By default it would be empty, meaning that the anchors are not loaded.

This should make it a bit more clear that one needs to pass anchors in order to use this annotator.
Otherwise another call `load_anchor` may be forgotten.

Reply:
Hmmm, adding the filename parameter adds a different inconsistency: what if someone constructs a `RowDiff<BRWT>` or a `RowDiff<RowSparse>` and passes in an anchor filename? These formats already have the anchors saved in the binary format itself.

How about we check if the anchors are loaded when the query methods (`get_xxx`) are called and print an error message that anchors should be loaded first? Currently I have an assert
`assert(anchor_.size() > 0 || diffs_.num_rows() == 0);`
but can make that stronger.

Reply:
Yeah, I see the inconsistency...

It's weird though that we do pass graph (which is always nullptr for `RowDiff<Column>`) but not anchors.

Maybe we'll revisit this in the next updates.